### PR TITLE
feat: Upgrading to angular 7.2.11 (#1524)

### LINKED ISSUE #1524: Upgrade to latest angular(-cli) and dependencies
- angular 7.2.11
- angular-cli 7.3.7
- rxjs 6.4.0
- ngrx 7.2
- any of the dev dependencies that we can update

### PR DESCRIPTION
**Pull request by hackergil**
Fixes #1524

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -121,7 +124,7 @@
     "karma-jasmine": "~1.1.2",
     "karma-jasmine-html-reporter": "^0.2.2",
     "karma-junit-reporter": "^1.2.0",
-    "ng-packagr": "^4.4.5",
+    "ng-packagr": "^4.0.0",
```
Comment (znikola) [inline package.json]: Why was this downgraded?

File: package.json [hunk 2]
```diff
@@ -57,44 +57,47 @@
...
-    "@angular/cli": "~7.2.1",
-    "@angular/compiler-cli": "7.2.0",
-    "@angular/language-service": "~7.2.0",
-    "@ngrx/store-devtools": "~7.0.0",
+    "@angular-devkit/build-angular": "~0.13.7",
+    "@angular-devkit/build-ng-packagr": "~0.13.7",
+    "@angular/cli": "~7.3.7",
+    "@angular/compiler-cli": "7.2.11",
+    "@angular/language-service": "~7.2.11",
+    "@ngrx/store-devtools": "~7.2.0",
```
Comment (znikola) [inline package.json]: Please upgrade to `7.4.0`

File: package.json [hunk 3]
```diff
@@ -57,44 +57,47 @@
...
-    "@angular/service-worker": "~7.2.0",
+    "@angular/router": "~7.2.11",
+    "@angular/service-worker": "~7.2.11",
     "@compodoc/compodoc": "^1.1.8",
     "@ng-bootstrap/ng-bootstrap": "^4.0.1",
     "@ng-select/ng-select": "^2.13.2",
-    "@ngrx/effects": "~7.0.0",
-    "@ngrx/router-store": "~7.0.0",
-    "@ngrx/store": "~7.0.0",
+    "@ngrx/effects": "~7.2.0",
```
Comment (znikola) [inline package.json]: Please upgrade all `ngrx` dependencies to `7.4.0` (see https://github.com/SAP/cloud-commerce-spartacus-storefront/issues/1524#issuecomment-478643141)

File: package.json [hunk 4]
```diff
@@ -134,6 +137,7 @@
     "tslib": "^1.9.3",
     "tslint": "~5.11.0",
     "typescript": "~3.2.2",
+    "webpack": "^4.0.0",
     "webpack-cli": "^3.1.2"
   }
 }
```
Comment (znikola) [inline package.json]: Please upgrade `karma` to `4.0.1`

File: package.json [hunk 5]
```diff
@@ -134,6 +137,7 @@
     "tslib": "^1.9.3",
     "tslint": "~5.11.0",
     "typescript": "~3.2.2",
+    "webpack": "^4.0.0",
```
Comment (znikola) [inline package.json]: In a newly CLI-generated angular project, there are no `webpack` and `webpack-cli` list as a dependency. Maybe we can remove it, and rely on angular to maintain and the most compatible version?

File: package.json (hunk 4 above)
Comment (znikola) [inline package.json]: There are some `devDependencies` that can be upgraded. You can use VSCode's built-in functionality to hover over a dependency version to see the most recent one. Please do this only for `devDependencies`

File: projects/core/package.json [hunk 6]
```diff
@@ -14,13 +14,13 @@
...
   "repository": "https://github.com/SAP/cloud-commerce-spartacus-storefront",
   "peerDependencies": {
-    "@angular/common": "~7.2.0",
-    "@angular/core": "~7.2.0",
-    "@ngrx/effects": "~7.0.0",
-    "@ngrx/router-store": "~7.0.0",
-    "@ngrx/store": "~7.0.0",
+    "@angular/common": "~7.2.11",
+    "@angular/core": "~7.2.11",
+    "@ngrx/effects": "~7.2.0",
```
Comment (znikola) [inline projects/core/package.json]: Please upgrade all ngrx dependencies to 7.4.0

File: projects/storefrontlib/package.json [hunk 7]
```diff
@@ -13,27 +13,27 @@
...
-    "@angular/platform-browser-dynamic": "~7.2.0",
+    "@angular/animations": "~7.2.11",
+    "@angular/common": "~7.2.11",
+    "@angular/compiler": "~7.2.11",
+    "@angular/core": "~7.2.11",
+    "@angular/forms": "~7.2.11",
+    "@angular/http": "~7.2.11",
+    "@angular/platform-browser": "~7.2.11",
+    "@angular/platform-browser-dynamic": "~7.2.11",
     "@angular/pwa": "^0.12.0",
```
Comment (znikola) [inline projects/storefrontlib/package.json]: Can be upgraded

File: projects/storefrontlib/package.json [hunk 8]
```diff
@@ -13,27 +13,27 @@
...
+    "@angular/router": "~7.2.11",
+    "@angular/service-worker": "~7.2.11",
     "@ng-bootstrap/ng-bootstrap": "^4.0.1",
     "@ng-select/ng-select": "^2.13.2",
-    "@ngrx/effects": "~7.0.0",
-    "@ngrx/router-store": "~7.0.0",
-    "@ngrx/store": "~7.0.0",
-    "@spartacus/core": "^0.1.0",
-    "@spartacus/styles": "^0.1.0",
+    "@ngrx/effects": "~7.2.0",
```
Comment (znikola) [inline projects/storefrontlib/package.json]: Please upgrade all ngrx dependencies to 7.4.0

File: package.json (hunk 5 above)
Comment (hackergil) [reply package.json]: I think SSR needs it 

> "build:ssr": "ng run storefrontapp:server && webpack --config projects/ssr/webpack.server.config.js --progress --colors",

Comment (znikola) [reply package.json]: Do you get an error if you remove it? It should still work, because we get `webpack` from angular.

File: projects/storefrontlib/karma.conf.js [hunk 9]
```diff
@@ -24,7 +24,7 @@ module.exports = function(config) {
         statements: 80,
         lines: 80,
         branches: 60,
-        functions: 80,
+        functions: 73,
```
Comment (znikola) [inline projects/storefrontlib/karma.conf.js]: Why was karma complaining about the coverage?

File: package.json [hunk 10]
```diff
@@ -57,52 +57,55 @@
...
-    "bootstrap": "^4.1.3",
+    "bootstrap": "^4.2.1",
     "core-js": "^2.5.7",
+    "express": "^4.15.2",
     "hamburgers": "^1.1.3",
     "i18next": "^15.0.6",
     "i18next-xhr-backend": "^2.0.1",
-    "ngrx-store-localstorage": "^5.1.0",
-    "rxjs": "^6.3.3",
+    "jquery": "1.9.1",
```
Comment (znikola) [inline package.json]: Please remove `jquery`. I believe it's needed for parts of bootstrap that we don't use.

File: package.json [hunk 11]
```diff
@@ -57,52 +57,55 @@
...
+    "ngrx-store-localstorage": "^7.0.0",
+    "rxjs": "^6.4.0",
+    "popper.js": "^1.14.6",
```
Comment (znikola) [inline package.json]: Why was this added?

File: package.json [hunk 12]
```diff
@@ -57,52 +57,55 @@
...
+    "@ngrx/effects": "~7.4.0",
+    "@ngrx/router-store": "~7.4.0",
+    "@ngrx/store": "~7.4.0",
     "@nguniversal/express-engine": "^7.0.2",
     "@nguniversal/module-map-ngfactory-loader": "^7.0.2",
     "@types/googlemaps": "^3.30.13",
-    "bootstrap": "^4.1.3",
+    "bootstrap": "^4.2.1",
     "core-js": "^2.5.7",
+    "express": "^4.15.2",
```
Comment (znikola) [inline package.json]: Why was this added?

File: projects/storefrontlib/karma.conf.js (hunk 9 above)
Comment (hackergil) [reply projects/storefrontlib/karma.conf.js]: Since we didn't touch any code I think it's just b/c of the version update

File: package.json (hunk 5 above)
Comment (hackergil) [reply package.json]: I get warnings if I remove it

File: package.json (hunk 10 above)
Comment (hackergil) [reply package.json]: We get warnings if we don't include it

File: package.json (hunk 11 above)
Comment (hackergil) [reply package.json]: Those are peer dependencies. We get warnings if we don't add them

File: package.json (hunk 10 above)
Comment (znikola) [reply package.json]: I believe that's fine, we had those warnings before.

File: package.json (hunk 11 above)
Comment (znikola) [reply package.json]: I believe that's fine, we had those warnings before.

File: package.json (hunk 5 above)
Comment (znikola) [reply package.json]: I believe that's fine, it should be installed by angular and put in `node_modules` to make it available for other packages
